mantidproject/mantid
# IN16B BATS energy transfer reduction

**Description of work.**
Implements the energy transfer reduction for the new BATS mode of the IN16B at the ILL.
This is the first cut tested on one example dataset, adjustments are foreseen once there is new data available.
Some cleanups and improvements are done in the loader as well.

**Report to:** M. Appel @ ILL

**To test:**
```
center, epp = IndirectILLEnergyTransfer(Run='215962', PulseChopper='34')
offset_p100 = IndirectILLEnergyTransfer(Run='215967', InputElasticChannelWorkspace=epp, PulseChopper='34')
offset_m275 = IndirectILLEnergyTransfer(Run='215968', InputElasticChannelWorkspace=epp, PulseChopper='34')

SumSpectra(InputWorkspace=center.getItem(0), EndWorkspaceIndex=2047, OutputWorkspace='sum_center')
SumSpectra(InputWorkspace=offset_m275.getItem(0), EndWorkspaceIndex=2047, OutputWorkspace='sum_m275')
SumSpectra(InputWorkspace=offset_p100.getItem(0), EndWorkspaceIndex=2047, OutputWorkspace='sum_p100')
```
Overplot the three `sum*` workspaces with log-y scale.
Elastic peak should be at 0 energy transfer, and the tunnel peaks should align nicely with each other.
The runs are in system test data, under ILL/IN16B. You might need to set the facility to ILL.

Fixes #25575 .

------------------------------

#### Reviewer ####
**It would be great to have this in nightly asap, hence high priority.**

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
